src/server/attestation/mod.rs

```diff
@@ -0,0 +1,330 @@
+            .unmatched_path
+            .first()
+            .and_then(|x| x.parse::<u32>().ok())
+            .ok_or_else(CoapError::not_found)?;
+
+        if request.unmatched_path.len() != 1 {
+            return Err(CoapError::not_found());
+        }
+
+        assert_content_format(&request.original, ContentFormat::ApplicationCBOR)?;
+        let evidence = &request.original.message.payload[..];
+        let mut client = client.lock().await;
+        let ac = client
+            .attestation
+            .acs
+            .get_mut(&ac_id)
+            .ok_or_else(CoapError::not_found)?;
+
+        let mut trussed = state.trussed.lock().await;
+        match verify_evidence(&mut *trussed, evidence, ac).await {
+            Ok(()) => {
+                info!("Attestation successful");
+                client.attestation.acs.remove(&ac_id);
+                let mut response = response_empty(&request);
+                response.set_status(ResponseType::Changed);
+                return Ok(response);
+            }
+            Err(e) => return Err(e),
+        }
+    }
+
+    assert_content_format(&request.original, ContentFormat::ApplicationCBOR)?;
+    verify_response_content_format(&request, ContentFormat::ApplicationCBOR)?;
+
+    // Treat lack of nonce as a signing error - client can't sign anything
+    // properly without asking for the nonce
+    let mut client = client.lock().await;
+    let nonce = client.nonce.take().ok_or_else(CoapError::forbidden)?;
+    let mut trussed = state.trussed.lock().await;
+    let hash = hash_signed_object(&mut *trussed, &request.original.message.payload)
+        .map_err(|()| CoapError::bad_request("Invalid CBOR"))?;
+
+    let aik = load_aik(&mut *trussed, &hash).map_err(|()| CoapError::not_found())?;
+    let meta: proto::Metadata =
+        decode_signed_cbor_req(&request.original, &mut *trussed, &aik, &nonce)
+            .map(|(meta, _)| meta)?;
+
+    info!("Attesting platform:");
+    info!("  MAC          : {}", meta.mac);
+    info!("  Manufacturer : {}", meta.manufacturer);
+    info!("  Product      : {}", meta.product_name);
+    info!("  Serial       : {}", meta.serial_number);
+
+    let rim = load_rim(&mut *trussed, &hash).map_err(|()| CoapError::not_found())?;
+    // TODO: policy should be loaded from
+    // internal storage
+    let policy = Policy::default();
+    let quote_nonce = generate_nonce(&mut *trussed);
+
+    let quote_request = trussed::cbor_serialize_bytes::<_, 512>(&proto::QuoteRequest {
+        nonce: proto::Nonce::new(&quote_nonce),
+        banks: policy.banks,
+    })
+    .map_err(|e| {
+        error!("CBOR serialization failed: {}", e);
+        CoapError::internal("Internal error")
+    })?;
+
+    // RNG locks Trussed, need to drop our lock now to prevent deadlock.
+    drop(trussed);
+    let mut response = create_object(
+        &request,
+        &mut client.attestation.acs,
+        AttestationContext {
+            rim,
+            nonce,
```

Comment:
This should be initialized to `quote_nonce`, right?

Reply:
Yes, it should - fixed.